### PR TITLE
Implement capsule library (tree_walker and unit tests)

### PR DESCRIPTION
**Pull request by theoreticalbts**

### CODE REVIEW

**Comment by theoreticalbts** [comment]
Fixed clang compilation errors, ready for review.

File: libraries/capsule/include/koinos/capsule/capsule.hpp [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+      }
+
+   private:
+      void _add_first_object(int64_t obj_index)
+      {
+         if( obj_index > 0 )
+         {
+            // Walk downward, going through the paths not taken.
+            int64_t leaf_id = obj_index << 1;    // Cannot overflow because of checks in add_object()
+            int64_t height = boost::multiprecision::detail::find_msb(leaf_id);
```
Comment (sgerbino) [inline libraries/capsule/include/koinos/capsule/capsule.hpp:139]: Applications should not be using `detail::` from boost.

File: tests/tests/capsule_test.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+   //                         /               \
+   //                 7                              23
+   //             /       \                       /      \
+   //         3              11              19              27
+   //       /   \          /    \          /    \          /    \
+   //     1       5       9      13      17      21      25      29
+   //    / \     / \     / \     / \     / \     / \     / \     / \
+   //   0   2   4   6   8  10  12   14 16   18 20   22 24   26 28   30
+
+   std::vector< std::tuple< int64_t, std::vector< int64_t >, std::string > > cases = {
```
Comment (sgerbino) [inline tests/tests/capsule_test.cpp:54]: We should avoid using nested vectors of magic numbers, it makes the test cases hard to understand and maintain.

**Review by sgerbino — CHANGES_REQUESTED** [review]

**Comment by mvandeberg** [comment]
Pull is closed due to age

**Pull request closed without merging**